google-cloud-spanner-cassandra/src/main/java/com/google/cloud/spanner/adapter/DriverConnectionHandler.java

```diff
@@ -270,4 +282,29 @@ private Optional<byte[]> prepareAttachmentForQueryId(
   private static String constructKey(byte[] queryId) {
     return PREPARED_QUERY_ID_ATTACHMENT_PREFIX + new String(queryId, StandardCharsets.UTF_8);
   }
+
+  private static ApiCallContext createContext(Frame frame) {
+    ApiCallContext context = GrpcCallContext.createDefault();
```

Comment:
Instead of creating instance (`GrpcCallContext.createDefault()`) each time, we can define the 2 class member (`private final GrpcCallContext defaultContext` and `private final GrpcCallContext defaultContextWithLAR`) and initialize them in constructor (`this.defaultContext = GrpcCallContext.createDefault()` and `this.defaultContextWithLAR = GrpcCallContext.createDefault().withExtraHeaders(..)`) and  use them here

```
if (isDML(..)) {
  return defaultContextWithLAR;
}

return defaultContext;
```

Reply:
done